components/camel-elytron/src/test/java/org/apache/camel/component/elytron/ElytronBearerTokenTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.elytron;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.util.Date;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import io.undertow.util.Headers;
+import org.apache.camel.CamelExecutionException;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.http.common.HttpOperationFailedException;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wildfly.security.WildFlyElytronBaseProvider;
+import org.wildfly.security.auth.realm.token.TokenSecurityRealm;
+import org.wildfly.security.auth.realm.token.validator.JwtValidator;
+import org.wildfly.security.authz.RoleDecoder;
+import org.wildfly.security.http.HttpConstants;
+import org.wildfly.security.http.bearer.WildFlyElytronHttpBearerProvider;
+
+public class ElytronBearerTokenTest extends BaseElytronTest {
+    private static final Logger LOG = LoggerFactory.getLogger(ElytronBearerTokenTest.class);
+
+    @Override
+    String getMechanismName() {
+        return HttpConstants.BEARER_TOKEN;
+    }
+
+    @Override
+    TokenSecurityRealm createBearerRealm() throws NoSuchAlgorithmException {
+        return TokenSecurityRealm.builder().principalClaimName("username")
+                .validator(JwtValidator.builder().publicKey(getKeyPair().getPublic()).build()).build();
+    }
+
+    @Override
+    WildFlyElytronBaseProvider getElytronProvider() {
+        return WildFlyElytronHttpBearerProvider.getInstance();
+    }
+
+    @Test
+    public void testBearerToken() throws Exception {
+        MockEndpoint mockEndpoint = getMockEndpoint("mock:myapp");
+        mockEndpoint.expectedHeaderReceived(Exchange.HTTP_METHOD, "POST");
+        LOG.debug("Number of exchanges in mock:myapp " + mockEndpoint.getExchanges().size());
```

Comment:
I see no route to `mock:myapp` and no assertions on mock. Can be these 3 lines removed (or extended routebuilder to actually fill that mock endpoint)? The same in the second test.